services/datamanager/datasync/upload_data_capture_file.go

```diff
@@ -26,23 +28,67 @@ func uploadDataCaptureFile(ctx context.Context, client v1.DataSyncServiceClient,
 		return nil
 	}
 
-	uploadMD := &v1.UploadMetadata{
-		PartId:           partID,
-		ComponentType:    md.GetComponentType(),
-		ComponentName:    md.GetComponentName(),
-		MethodName:       md.GetMethodName(),
-		Type:             md.GetType(),
-		MethodParameters: md.GetMethodParameters(),
-		FileExtension:    md.GetFileExtension(),
-		Tags:             md.GetTags(),
+	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && len(sensorData) > 1 {
+		return errors.New("binary sensor data file with more than one sensor reading is not supported")
 	}
 
-	// If it's a large binary file, we need to upload it in chunks.
-	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && f.Size() > MaxUnaryFileSize {
-		if len(sensorData) > 1 {
-			return errors.New("binary sensor data file with more than one sensor reading is not supported")
+	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && md.GetMethodName() == "GetImages" {
```

Comment:
So that this doesn't break if the collector string changes, would export Method and GetImages in camera/collectors.go and use GetImages.String() here

Reply:
Done.